### PR TITLE
Initial version.

### PR DESCRIPTION
**Pull request by smarnach**
This is the first version of an Ansible role to deploy a Mattermost server for OpenCraft.  For testing instructions see https://github.com/open-craft/deployment-secrets-services/pull/31.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Ansible role for OpenCraft's internal Mattermost chat server
+============================================================
+
+This role sets up a Mattermost server.  The service is run behing nginx, and
```
Comment (tomaszgy) [inline README.md]: Nit: did you mean ..._behind_ nginx?

File: tasks/nginx.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+- name: install certbot apt package
+  apt:
+    update_cache: yes
+    name: python-certbot-nginx
+
+# We need to stop nginx before requesting a certificate, so we can use the Certbot standalone
+# server.  Using the nginx plugin of Certbot is difficult, because it would compete with Ansible
+# for the ownership of the nginx site configuration file.  nginx will be restarted at the end of
+# the file, so we don't need to start it here.  This task will only be executed when no SSL cert
```
Comment (tomaszgy) [inline tasks/nginx.yml:27]: Just out of curiosity: how is that assured that the task will only be executed when no SSL cert exists yet? :)

File: tasks/mattermost.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    dest: "{{ MATTERMOST_VERSION_FILE }}"
+    content: "{{ MATTERMOST_VERSION }}"
+    owner: mattermost
+    group: mattermost
+  register: mattermost_version
+
+# This block is only run when either the previous installation attempt wasn't completed or a new
+# version of Mattermost needs to be installed.  In the latter case it performs a fully automated
+# upgrade.
+- name: install new version of Mattermost
```
Comment (tomaszgy) [inline tasks/mattermost.yml:56]: UPDATE after running `ansible-playbook`:
this line throws an error: `'name' is not a valid attribute for a Block` when using [Ansible 2.0.1](https://github.com/open-craft/deployment-deploy-services/blob/smarnach/mattermost/requirements.txt) (I installed it through `deploy/requirements.txt`). The`name` attribute is [introduced in Ansible 2.3](http://docs.ansible.com/ansible/latest/playbooks_blocks.html#id1) so I guess either`requirements.txt` should be updated or the `name` attribute removed.

File: README.md (hunk 1 above)
Comment (smarnach) [reply README.md]: Thanks, fixed.

File: tasks/mattermost.yml (hunk 3 above)
Comment (smarnach) [reply tasks/mattermost.yml:56]: @tomaszgy Oh, right, we should update the Ansible version.  I actually did use version 2.3.2.  I'll include the version update in https://github.com/open-craft/deployment-deploy-services/pull/18.

File: tasks/nginx.yml (hunk 2 above)
Comment (smarnach) [reply tasks/nginx.yml:27]: @tomaszgy The `creates` argument below tells Ansible what file is created by the command, and only runs the command if the file does not exist yet.  Certbot creates multiple files, but specifying any one of them should work fine.

Comment (tomaszgy) [reply tasks/nginx.yml:27]: @smarnach, thanks for explaining it to me! I'm not very good with Ansible but if the `creates` trick is not part of Ansible common knowledge, maybe we could expand the comment a bit here? :) If it is something well-known though then all's good.

**Comment by tomaszgy** [comment]
👍

File: tasks/nginx.yml (hunk 2 above)
Comment (smarnach) [reply tasks/nginx.yml:27]: @tomaszgy While not everyone might know this, it's not an obscure trick – it's the only purpose of the `creates` argument.  I've added a hint to the comment, though.

Comment (tomaszgy) [reply tasks/nginx.yml:27]: @smarnach, thanks! (Also for having a chance to learn few things while reviewing :))

**Pull request merged**